added -h flag

AzureLeMoon opened the pull request:
added the -h flag to listen on 0.0.0.0 instead of localhost.

the goal is to make this easier to use by setting it up on an Iran Based VPS to connect users to warp using a simple socks proxy which is much easier to deploy especially on IOS.

uoosef commented:
thanx, for pr unfortunately i changed main.go befor seeing your pr please update pr based on new main.go version

AzureLeMoon commented:
> thanx, for pr unfortunately i changed main.go befor seeing your pr please update pr based on new main.go version

hey , there doesn't seem to be any issues now.
also i've added the -p flag for setting the port along with the host  along with the description on the readme file.
should be good to merge

X-Oracle commented:
usually `-h` is used to echo help.
wouldn't it better to use `-g` `--global` for 0.0.0.0 ?

AzureLeMoon commented:
> usually `-h` is used to echo help. wouldn't it better to use `-g` `--global` for 0.0.0.0 ?

That is correct however i named it `-h` for host since i changed it from a bool value to string for user to specify the IP address they want to use instead of just `localhost` or `0.0.0.0`.
e.g. 10.10.10.10:[port] to bind to a single remotely accessible IP address

X-Oracle commented:
That's a good point.

Tho I would prefer `-l` and `--listen` for custom IPs.

Any ways thanks for the commit.
Best of luck.

hatinati2 commented:
سلام وقت بخیر
ممنون از زحمات شما
آیا امکانش هست این اسکریپت روی سرور اوبونتو ایران نصب کرد و بدون نیاز به سرور خارج برای x-ui  استفاده کرد؟

AzureLeMoon commented:
> سلام وقت بخیر ممنون از زحمات شما آیا امکانش هست این اسکریپت روی سرور اوبونتو ایران نصب کرد و بدون نیاز به سرور خارج برای x-ui استفاده کرد؟

سلام  , آره کافیه پورت و آیپیش رو (علاوه بر بقیه فلگ ها) مشخص کنید و روی به صورت سرویس اجراش کنید بعد به عنوان Outbound توی x-ui تعریفش کنید.
اگرچه که به نظرم از داخل سرور نگیرید بهتره درکل.

shahryarjb commented:
> > سلام وقت بخیر ممنون از زحمات شما آیا امکانش هست این اسکریپت روی سرور اوبونتو ایران نصب کرد و بدون نیاز به سرور خارج برای x-ui استفاده کرد؟
> 
> سلام , آره کافیه پورت و آیپیش رو (علاوه بر بقیه فلگ ها) مشخص کنید و روی سرور به صورت سرویس اجراش کنید بعد به عنوان Outbound توی x-ui تعریفش کنید. اگرچه که به نظرم از داخل سرور نگیرید بهتره درکل.

درود خدمت شما تشکر از توضیحات. جسارتا چطور می شه جلوشو گرفت مستقیم به پرت خود وارپ کسی وصل نشه حتما از خود v2ray که براش یوزر می سازیم بگیره؟

AzureLeMoon commented:
> > > سلام وقت بخیر ممنون از زحمات شما آیا امکانش هست این اسکریپت روی سرور اوبونتو ایران نصب کرد و بدون نیاز به سرور خارج برای x-ui استفاده کرد؟
> > 
> > 
> > سلام , آره کافیه پورت و آیپیش رو (علاوه بر بقیه فلگ ها) مشخص کنید و روی سرور به صورت سرویس اجراش کنید بعد به عنوان Outbound توی x-ui تعریفش کنید. اگرچه که به نظرم از داخل سرور نگیرید بهتره درکل.
> 
> درود خدمت شما تشکر از توضیحات. جسارتا چطور می شه جلوشو گرفت مستقیم به پرت خود وارپ کسی وصل نشه حتما از خود v2ray که براش یوزر می سازیم بگیره؟

تو تنظیمات پیشفرض ازونجایی که خودش روی 127.0.0.1 سوار میشه لازم نیست کار خاصی بکنید.

AzureLeMoon commented:
Closing this since you can use the bindaddress flag for the same purpose.
